tools/testing/target_determination/heuristics/public_bindings.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from logging import warn
+from typing import Any
+
+from tools.testing.target_determination.heuristics.interface import (
+    HeuristicInterface,
+    TestPrioritizations,
+)
+from tools.testing.target_determination.heuristics.utils import (
+    query_changed_files,
+)
+
+from tools.testing.test_run import TestRun
+
+
+class PublicBindings(HeuristicInterface):
+    # Literally just a heuristic for test_public_bindings.  Pretty much anything
+    # that changes the public API can affect this test
+    test_public_bindings = "test_public_bindings"
+
+    def __init__(self, **kwargs: dict[str, Any]) -> None:
+        super().__init__(**kwargs)
+
+    def get_prediction_confidence(self, tests: list[str]) -> TestPrioritizations:
+        test_ratings = {}
+        try:
+            changed_files = query_changed_files()
+        except Exception as e:
+            warn(f"Can't query changed test files due to {e}")
+            changed_files = []
+
+        if any(file.startswith("torch/") for file in changed_files):
```

Comment:
I guess we don't care about .cpp file changes
```suggestion
        if any(file.startswith("torch/") and file.endswith(".py") for file in changed_files):
```